ArduCopter/GCS_Mavlink.cpp

```diff
@@ -1077,16 +1086,8 @@ void GCS_MAVLINK_Copter::handleMessage(const mavlink_message_t &msg)
             }
         }
 
-        // if the body_yaw_rate field is ignored, use the commanded yaw position
-        // otherwise use the commanded yaw rate
-        bool use_yaw_rate = false;
-        if ((packet.type_mask & (1<<2)) == 0) {
-            use_yaw_rate = true;
-        }
-
-        copter.mode_guided.set_angle(Quaternion(packet.q[0],packet.q[1],packet.q[2],packet.q[3]),
-                climb_rate_or_thrust, use_yaw_rate, packet.body_yaw_rate, use_thrust);
-
+        copter.mode_guided.set_angle(Quaternion(packet.q[0],packet.q[1],packet.q[2],packet.q[3]), climb_rate_or_thrust, use_yaw_rate, packet.body_yaw_rate, use_thrust);
+        copter.mode_guided.set_rate_bf_roll_pitch_yaw(packet.body_pitch_rate, packet.body_roll_rate, packet.body_yaw_rate, use_roll_pitch_rates);
```

Comment:
This can set both attitude and rates even if we were told to ignore for example rates. Example, `packet.body_pitch_rate` could contain a value  and now we are sending that to the controller even though we were told to ignore it. 

See `SET_POSITION_TARGET_LOCAL_NED` below for how it builds up targets from the typemasks given and sends to the appropriate controller.